### ISSUE TITLE
Terraform Crash Error: rpc error: code = Unavailable desc = transport is closing

### ISSUE THREAD

**Issue by Geartrixy** [hashicorp/terraform#22349]
### Terraform Version / AWS Provider Version

```
0.12.6 / 2.22.0
```

### Terraform Configuration Files (Child Module)

```hcl
resource "aws_lb_listener" "http" {
  count             = (var.lb_http_listener == true ? 1 : 0)
  load_balancer_arn = aws_lb.application_loadbalancer.arn
  port              = var.lb_http_listener_port
  protocol          = "HTTP"

  default_action {
    target_group_arn = aws_lb_target_group.tg_http[0].arn
    type             = "forward"
  }
}

# Create http listener rules
resource "aws_lb_listener_rule" "http" {
  count              = (var.lb_http_listener == true ? length(var.host_headers) == length(var.http_target_group_names) ? length(var.host_headers) : 0 : 0)
  listener_arn       = aws_lb_listener.http[0].arn

  action {
    type             = "forward"
    target_group_arn = element(aws_lb_target_group.tg_http.*.arn, count.index)
  }

  condition {
    field            = "host-header"
    values           = [element(var.host_headers, count.index)]
  }
}

resource "aws_lb_listener" "https" {
  count              = var.lb_https_listener == true ? 1 : 0
  load_balancer_arn  = aws_lb.application_loadbalancer.arn
  port               = var.lb_https_listener_port
  protocol           = "HTTPS"

  default_action {
    target_group_arn = var.lb_https_offloading == true ? aws_lb_target_group.tg_http[0].arn : aws_lb_target_group.tg_https[0].arn
    type             = "forward"
  }

  certificate_arn    = var.certificate_arn
}

# Create https listener rules
resource "aws_lb_listener_rule" "https" {
  count              = (var.lb_https_listener == true ? length(var.host_headers) == length(var.https_target_group_names) ? length(var.host_headers) : 0 : 0)
  listener_arn       = aws_lb_listener.https[0].arn

  action {
    type             = "forward"
    target_group_arn = var.lb_https_offloading == true ?  element(aws_lb_target_group.tg_http.*.arn, count.index) : element(aws_lb_target_group.tg_https.*.arn, count.index)
  }

  condition {
    field            = "host-header"
    values           = [element(var.host_headers, count.index)]
  }
}

resource "aws_lb" "application_loadbalancer" {
  ### Required Arguments ###
  name               = "${var.prefix}${var.lb_name}${var.suffix}"
  internal           = var.lb_internal
  load_balancer_type = "application"
  # at least 2 subnets must be specified
  subnets            = local.lb_subnet_ids
  # if no security group is specified, "default" will be assigned
  security_groups    = concat(local.lb_security_groups)

  idle_timeout               = var.lb_idle_timeout
  enable_http2               = var.lb_enable_http2
  enable_deletion_protection = var.lb_enable_deletion_protection
}

resource "aws_lb_target_group" "tg_http" {
  # Check the number of http target group names matches the number of http target group ports.
  # If check is ok creates the number of http target group resources based on the number of http target group names
  count = var.lb_http_listener == true ? length(var.http_target_group_names) == length(var.http_target_group_ports) ? length(var.http_target_group_names) : 0 : 0

  ### Required Arguments ###
  name     = "${var.prefix}${element(var.http_target_group_names, count.index)}${var.suffix}" # default prefix/suffix = "". Default target group name = ["http-target-group"] N.B. 32 Character limit with prefix/suffix
  port     = element(var.http_target_group_ports, count.index)                                # default = [80]
  protocol = "HTTP"
  vpc_id   = var.vpc_id

  health_check {
    interval            = var.http_health_check_interval            #default = 30
    path                = var.http_health_check_path                #default = "/"
    port                = var.http_health_check_port                #default = "traffic-port"
    protocol            = "HTTP"
    timeout             = var.http_health_check_timeout             #default = 5
    healthy_threshold   = var.http_health_check_healthy_threshold   #default = 3
    unhealthy_threshold = var.http_health_check_unhealthy_threshold #default = 3
    matcher             = var.http_health_check_matcher             #default = 200
  }

  target_type = "instance"

  ### Optional Arguments ###
  deregistration_delay = var.http_target_group_deregistration_delay # default = 300.

  stickiness {
    enabled         = var.http_target_group_stickiness_enabled         # default set to false
    type            = "lb_cookie"                                      # default = lb_cookie. Only possible value for applicatoin load balancer
    cookie_duration = var.http_target_group_stickiness_cookie_duration # default 8640 seconds (1 day).
  }
  #tags = https://github.com/hashicorp/terraform/issues/15226

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_lb_target_group" "tg_https" {
  # Check the number of https target group names matches the number of https target group ports.
  # If check is ok creates the number of https target group resources based on the number of https target group names
  count = var.lb_https_listener == true ? var.lb_https_offloading == false ? length(var.https_target_group_names) == length(var.https_target_group_ports) ? length(var.https_target_group_names) : 0 : 0 : 0

  ### Required Arguments ###
  name     = "${var.prefix}${element(var.https_target_group_names, count.index)}${var.suffix}" # default prefix/suffix = "". Default target group name = ["https-target-group"] N.B. 32 Character limit with prefix/suffix
  port     = element(var.https_target_group_ports, count.index)                                # default = [443]
  protocol = "HTTPS"
  vpc_id   = var.vpc_id

  health_check {
    interval            = var.https_health_check_interval #default = 30
    path                = var.https_health_check_path     #default = "/"
    port                = var.https_health_check_port     #default = "traffic-port"
    protocol            = "HTTPS"
    timeout             = var.https_health_check_timeout             #default = 5
    healthy_threshold   = var.https_health_check_healthy_threshold   #default = 3
    unhealthy_threshold = var.https_health_check_unhealthy_threshold #default = 3
    matcher             = var.https_health_check_matcher             #default = 200
  }

  target_type = "instance"

  ### Optional Arguments ###
  deregistration_delay = var.https_target_group_deregistration_delay # default = 300.

  stickiness {
    enabled         = var.https_target_group_stickiness_enabled         # default set to false
    type            = "lb_cookie"                                       # default = lb_cookie. Only possible value for applicatoin load balancer
    cookie_duration = var.https_target_group_stickiness_cookie_duration # default 8640 seconds (1 day).
  }

  lifecycle {
    create_before_destroy = true
  }
}

locals {
lb_security_groups = (var.lb_http_listener == true ? var.lb_https_listener == true ? concat(var.http_security_group_ids, var.https_security_group_ids) : var.http_security_group_ids : var.https_security_group_ids)

  lb_subnet_ids =     (var.lb_internal == true ? var.lb_private_subnet_ids : var.lb_public_subnet_ids)
}

### Variables ### 

data "aws_caller_identity" "current" {
}

# Account
provider "aws" {
  region = var.aws_region
}

# Region
variable "aws_region" {
  description = "The AWS region to run in."
  default     = "eu-central-1"
}

# VPC ID
variable "vpc_id" {
  description = "The VPC ID in which the resources should be created."
}

# Prefix
variable "prefix" {
  description = "A prefix which is added to each resource name."
  default     = ""
}

# Suffix
variable "suffix" {
  description = "A suffix which is added to each resource name."
  default     = ""
}
# HTTP target group names
variable "http_target_group_names" {
  type        = list(string)
  description = "Name(s) of the http target group(s)"
  default     = ["http-target-group"] #N.B. Target Group name acceptable characters: letters, digits or the dash
}

# HTTP target group ports
variable "http_target_group_ports" {
  type        = list(string)
  description = "Port(s) of the http target group(s)"
  default     = [80]
}

# HTTP target group degregistration delay
variable "http_target_group_deregistration_delay" {
  description = "Deregistration delay for the http target group(s)"
  default     = 300 #Possible values in seconds: 0-3600
}

##HTTP target group health checks
# HTTP target group health check intveral
variable "http_health_check_interval" {
  description = "Set HTTP health check interval"
  default     = 30 #min=5 max=300
}

# HTTP target group health check path
variable "http_health_check_path" {
  description = "Set HTTP health check path"
  default     = "/"
}

# HTTP target group health check port
variable "http_health_check_port" {
  description = "Set HTTP health check port"
  default     = "traffic-port"
}

# HTTP target group health check protocol
variable "http_health_check_protocol" {
  description = "Set HTTP health check protocol"
  default     = "HTTP"
}

# HTTP target group health check timeout
variable "http_health_check_timeout" {
  description = "Set HTTP health check timeout"
  default     = 5 #min=2 max=60
}

# HTTP target group health check healthy threshold
variable "http_health_check_healthy_threshold" {
  description = "Set HTTP health check healthy threshold"
  default     = 3
}

# HTTP target group health cehck unhealthy threshold
variable "http_health_check_unhealthy_threshold" {
  description = "Set HTTP health check unhealthy threshold"
  default     = 3
}

# HTTP target group health check  matcher
variable "http_health_check_matcher" {
  description = "Set HTTP health check matcher"
  default     = 200 #Matcher aka success code. Range-example: 200-299
}

## HTTP target group sticky cookie settings
# HTTP target group stickiness enabled/disabled
variable "http_target_group_stickiness_enabled" {
  description = "Turn stickiness on/off for the http target group(s)"
  default     = false
}

# HTTP cookie duration for stickiness if enabled
variable "http_target_group_stickiness_cookie_duration" {
  description = "Cookie Duration for stickiness of the http target group(s)"
  default     = 8640 #Possible values 1-604800 (604800 = 1 day).
}
### HTTPS target group variables ###
# HTTPS target group names
variable "https_target_group_names" {
  type        = list(string)
  description = "Name(s) of the https target group(s)"
  default     = ["https-target-group"] #N.B. Target Group name acceptable characters: letters, digits or the dash
}

# HTTPS target group ports
variable "https_target_group_ports" {
  type        = list(string)
  description = "Port(s) of the https target group(s)"
  default     = [443]
}

# HTTPS target group degregistration delay
variable "https_target_group_deregistration_delay" {
  description = "Deregistration delay for the https target group(s)"
  default     = 300 #Possible values in seconds: 0-3600
}

##HTTPS target group health checks
# HTTPS target group health check intveral
variable "https_health_check_interval" {
  description = "Set HTTPS health check interval"
  default     = 30 #min=5 max=300
}

# HTTPS target group health check path
variable "https_health_check_path" {
  description = "Set HTTPS health check path"
  default     = "/"
}

# HTTPS target group health check port
variable "https_health_check_port" {
  description = "Set HTTPS health check port"
  default     = "traffic-port"
}

# HTTPS target group health check protocol
variable "https_health_check_protocol" {
  description = "Set HTTPS health check protocol"
  default     = "HTTPS"
}

# HTTPS target group health check timeout
variable "https_health_check_timeout" {
  description = "Set HTTPS health check timeout"
  default     = 5 #min=2 max=60
}

# HTTPS target group health check healthy threshold
variable "https_health_check_healthy_threshold" {
  description = "Set HTTPS health check healthy threshold"
  default     = 3
}

# HTTPS target group health cehck unhealthy threshold
variable "https_health_check_unhealthy_threshold" {
  description = "Set HTTPS health check unhealthy threshold"
  default     = 3
}

# HTTPS target group health check  matcher
variable "https_health_check_matcher" {
  description = "Set HTTPS health check matcher"
  default     = 200 #Matcher aka success code. Range-example: 200-299
}

## HTTPS target group sticky cookie settings
# HTTPS target group stickiness enabled/disabled
variable "https_target_group_stickiness_enabled" {
  description = "Turn stickiness on/off for the https target group(s)"
  default     = false
}

# HTTPS cookie duration for stickiness if enabled
variable "https_target_group_stickiness_cookie_duration" {
  description = "Cookie Duration for stickiness of the https target group(s)"
  default     = 8640 #Possible values 1-604800 (604800 = 1 day).
}

# Load balancer idle timeout
variable "lb_idle_timeout" {
  description = "Idle timeout for the loadbalancer."
  default     = 60 #seconds
}

# Load balancer enable http2
variable "lb_enable_http2" {
  description = "Define is http2 is enabled"
  default     = true
}

# Load balancer deletion protection
variable "lb_enable_deletion_protection" {
  description = "Enable deletion protection"
  default     = false
}

### internal / external
# Load balancer name
variable "lb_name" {
  description = "Name of the loadbalancer"
  default     = "loadbalancer"
}

# Load balancer internal or external
variable "lb_internal" {
  description = "Define if the loadbalancer should be internal or not"
  default     = true
}

## Load balancer listener variables
# Load balancer HTTP listener needed?
variable "lb_http_listener" {
  description = "If true add a HTTP listener"
  default     = false
}

# Load balancer HTTPS listener needed?
variable "lb_https_listener" {
  description = "If true add a HTTPS listener"
  default     = false
}

# Load balancer HTTPS offloading?
variable "lb_https_offloading" {
  description = "If true offload to HTTP"
  default     = true
}

# Load balancer HTTP listener port
variable "lb_http_listener_port" {
  description = "HTTP listener port of the loadbalancer"
  default     = 80
}

# Loadbalancer HTTPS listener port
variable "lb_https_listener_port" {
  description = "HTTPS listener port of the loadbalancer"
  default     = 443
}

## host headers
# host headers for load balancer listener rules
variable "host_headers" {
  type        = list(string)
  description = "Host headers for load balancer listener rules"
  #default     = []
}

# Certificate ARN for the HTTPS listener
variable "certificate_arn" {
  description = "Certificate ARN for the HTTPS listener"
  default     = null
}


## subnet groups
# Load balancer public subnet groups
variable "lb_public_subnet_ids" {
  type        = list(string)
  description = "The Public Subnet ID(s) which should be attached to the loadbalancer"
  default     = null
}

# Load balancer private subnet groups
variable "lb_private_subnet_ids" {
  type        = list(string)
  description = "The Private Subnet ID(s) which should be attached to the loadbalancer"
  default     = null
}

## Load balancer security groups
# Load balancer additional security groups

variable "http_security_group_ids" {
  type        = list(string)
  description = "The HTTP security groups for the load balancer"
  default     = [null]
}

variable "https_security_group_ids" {
  type        = list(string)
  description = "The HTTPS security groups for the load balancer"
  default     = [null]
}

variable "target_ids" {
  description = "Instance IDs for the target group(s)"
  default     = ""
}

```
### Terraform Configuration Files (Root Module)
```hcl


module "alb_ext" {
  source = "../terraform-aws-alb"

  # network
  aws_region            = "eu-central-1"
  vpc_id                = "vpc-xxx"
  lb_public_subnet_ids   = var.public_sn_ids
  target_ids            = "i-xxx"   

 # prefix and suffix
  prefix      = "D-Pre-"
  suffix      = ""

  # load balancer
  lb_name             = "Loadbalancer-Ext-euw"
  lb_internal         = false
  ## listeners
  lb_http_listener    = true
  lb_https_listener   = true
  lb_https_offloading = true

  host_headers = var.ext_host_headers
  certificate_arn  = var.certificate_arn
    

  

  # target groups
  http_target_group_names = var.tg_http_names
  http_target_group_ports = var.tg_http_ports
  #
  http_health_check_interval            = 30                      #default = 30
  http_health_check_path                = "/status/load-balancer" #default = "/"
  http_health_check_port                = "traffic-port"          #default = "traffic-port"
  http_health_check_timeout             = 10                      #default = 5
  http_health_check_healthy_threshold   = 5                       #default = 3
  http_health_check_unhealthy_threshold = 3                       #default = 3
  http_health_check_matcher             = 200                     #default = 200 (Success codes)
  #
  http_target_group_deregistration_delay       = 300   #default = 300 (seconds)
  http_target_group_stickiness_enabled         = false #default set to false
  http_target_group_stickiness_cookie_duration = 8640  #default 8640 seconds (1 day)



  https_health_check_interval            = 30                            #default = 30
  https_health_check_path                = "/status/load-balancer"       #default = "/"
  https_health_check_port                = 1500                          #default = "traffic-port"
  https_health_check_timeout             = 10                            #default = 5
  https_health_check_healthy_threshold   = 5                             #default = 3
  https_health_check_unhealthy_threshold = 3                             #default = 3
  https_health_check_matcher             = 200                           #default = 200 (Success codes)

  # HTTPS optional arguments:

  https_target_group_deregistration_delay       = 300   #default = 300 (seconds)
  https_target_group_stickiness_enabled         = false #default set to false
  https_target_group_stickiness_cookie_duration = 8640  #default 8640 seconds (1 day)
 
}

module "alb_int" {
  source = "../terraform-aws-alb"


  # network
  aws_region            = "eu-central-1"
  vpc_id                = "vpc-xxx"
  lb_private_subnet_ids = var.private_sn_ids
  target_ids            = "i-xxx"   

  # prefix and suffix
  prefix      = "D-Pre-"
  suffix      = ""
  
  # load balancer
  lb_name             = "Loadbalancer-Int-euw"
  lb_internal         = true
  ## listeners
  lb_http_listener    = true
  lb_https_listener   = true
  lb_https_offloading = false
  host_headers        = var.int_host_headers
  certificate_arn     = var.certificate_arn

  
 

   

  # target groups
  http_target_group_names = var.tg_http_names
  http_target_group_ports = var.tg_http_ports

  #
  http_health_check_interval            = 30                      #default = 30
  http_health_check_path                = "/status/load-balancer" #default = "/"
  http_health_check_port                = "traffic-port"          #default = "traffic-port"
  http_health_check_timeout             = 10                      #default = 5
  http_health_check_healthy_threshold   = 5                       #default = 3
  http_health_check_unhealthy_threshold = 3                       #default = 3
  http_health_check_matcher             = 200                     #default = 200 (Success codes)

  #
  http_target_group_deregistration_delay       = 300   #default = 300 (seconds)
  http_target_group_stickiness_enabled         = false #default set to false
  http_target_group_stickiness_cookie_duration = 8640  #default 8640 seconds (1 day)


  # HTTPS optional arguments:

  https_target_group_deregistration_delay       = 300   #default = 300 (seconds)
  https_target_group_stickiness_enabled         = false #default set to false
  https_target_group_stickiness_cookie_duration = 8640  #default 8640 seconds (1 day)
 
}

### terraform.tfvars ###
tg_http_names = ["App1", "App2"]

tg_https_names = ["App1", "App2"]

tg_http_ports = ["10000", "10001"]

tg_https_ports = ["443", "443"]

int_host_headers = ["app1.mywebsite.com", "app2.mywebsite.com"]

ext_host_headers = ["app1.mywebsite.co.uk", "app2.mywebsite.co.uk"]

private_sn_ids = ["subnet-xxx", "subnet-xxx"]

public_sn_ids = ["subnet-xxx", "subnet-xxx"]

certificate_arn = "arn:aws:acm:eu-central-1:xxx:certificate/xxx"

http_security_group_ids = ["sg-xxx", "sg-xxx"]

https_security_group_ids = ["sg-xxx", "sg-xxx"]
```
### Debug Output
https://gist.github.com/Geartrixy/d4de4f5b088042d2227d1754217981db

### Crash Output
https://gist.github.com/Geartrixy/f8070f36203185cd2cdc0edb93b918b2

### Steps to Reproduce
Terraform init is successful.
Terraform apply crashes.

### Additional Context
Some resource IDs/Content omitted with "xxx". The proper values are used with the apply.

**Comment by bflad** [comment]
Stacktrace from `helper/schema`:

```
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 52 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0x42ada20, 0x3e80b40)
 /opt/goenv/versions/1.12.6/src/reflect/value.go:1813 +0x170
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validatePrimitive(0xc00058fb30, 0xc000bef500, 0x11, 0x0, 0x0, 0xc000650300, 0xc00098ccf0, 0x1, 0x3e18bc0, 0xc0004f2e60, ...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1694 +0xcf
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateType(0xc00058fb30, 0xc000bef500, 0x11, 0x0, 0x0, 0xc000650300, 0xc00098ccf0, 0x4089e3, 0xc000016000, 0x44a4040, ...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1779 +0x576
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateList(0xc00058fb30, 0x4ca0ded, 0xf, 0x3e18bc0, 0xc0004f2e60, 0xc000651100, 0xc00098ccf0, 0x0, 0x42b6300, 0xc00098cd20, ...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1495 +0x873
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateType(0xc00058fb30, 0x4ca0ded, 0xf, 0x3e18bc0, 0xc0004f2e60, 0xc000651100, 0xc00098ccf0, 0x42b6300, 0xc00098cd20, 0x0, ...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1775 +0xa5
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validate(0xc00058fb30, 0x4ca0ded, 0xf, 0xc000651100, 0xc00098ccf0, 0x1, 0xc000b7b7e8, 0x4eaaab, 0xc000982000, 0xc0006aa000, ...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1382 +0x21a
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.validateObject(0xc00058fb30, 0x0, 0x0, 0xc00058fb30, 0xc00098ccf0, 0x4c8735d, 0xc000151456, 0x6, 0xc000402501, 0x1700000000000000, ...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:1661 +0x1cd
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.schemaMap.Validate(...)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/schema.go:677
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Validate(0xc00063f180, 0xc00098ccf0, 0xc000151456, 0x6, 0xc00066afb8, 0x1, 0x40, 0x0)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:357 +0x65
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).ValidateResource(0xc000660400, 0xc000151456, 0x6, 0xc00098ccf0, 0xc000b95f70, 0xc00098ccf0, 0xc0009aea58, 0x42b6300, 0xc00098cc90, 0x0)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:242 +0x1df
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/plugin.(*GRPCProviderServer).ValidateResourceTypeConfig(0xc000154860, 0x552ea40, 0xc000b83c20, 0xc000995380, 0xc000154860, 0xc000b83c20, 0xc00096fbd0)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/plugin/grpc_provider.go:206 +0x1e0
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/internal/tfplugin5._Provider_ValidateResourceTypeConfig_Handler(0x4b77f80, 0xc000154860, 0x552ea40, 0xc000b83c20, 0xc0000c0f50, 0x0, 0x552ea40, 0xc000b83c20, 0xc0006b8180, 0x17b)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/internal/tfplugin5/tfplugin5.pb.go:3109 +0x245
github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc000071500, 0x5550ec0, 0xc000520300, 0xc00061ea00, 0xc000657350, 0x93919d0, 0x0, 0x0, 0x0)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:966 +0x477
github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).handleStream(0xc000071500, 0x5550ec0, 0xc000520300, 0xc00061ea00, 0x0)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:1245 +0xd2c
github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000150660, 0xc000071500, 0x5550ec0, 0xc000520300, 0xc00061ea00)
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:685 +0xa6
created by github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
 /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:683 +0xa8
```

**Comment by teamterraform** [comment]
Hi @Geartrixy,

Thanks for reporting the issue. 
This is fixed in the current Terraform master branch, and will be available when the aws provider updates to the latest release.

**Comment by bflad** [comment]
Looks like the fix was part of v0.12.6 of the Terraform Provider SDK, which was included with today's release of version 2.23.0 of the Terraform AWS Provider. 👍

**Comment by Geartrixy** [comment]
Thanks! :D
